src/sage/matroids/chow_ring_ideal.py

```diff
@@ -0,0 +1,798 @@
+             Ac*Ad, Ac*Ade, Ac*Af, Ac*Aabf, Ac*Adf, Ac*Aef, Ac*Ag, Ac*Aadg,
+             Ac*Abeg, Ad*Ae, Ae*Abcd, Ae*Af, Ae*Aabf, Ae*Adf, Ae*Ag, Ae*Aadg,
+             Ae*Acfg, Ad*Aace, Aace*Ade, Aace*Abcd, Af*Aace, Aabf*Aace,
+             Aace*Adf, Aace*Aef, Ag*Aace, Aace*Aadg, Aace*Abeg, Aace*Acfg,
+             Ad*Af, Ad*Aabf, Ad*Aef, Ad*Ag, Ad*Abeg, Ad*Acfg, Abcd*Ade, Af*Ade,
+             Aabf*Ade, Ade*Adf, Ade*Aef, Ag*Ade, Aadg*Ade, Abeg*Ade, Acfg*Ade,
+             Af*Abcd, Aabf*Abcd, Abcd*Adf, Abcd*Aef, Ag*Abcd, Aadg*Abcd,
+             Abcd*Abeg, Abcd*Acfg, Af*Ag, Af*Aadg, Af*Abeg, Aabf*Adf, Aabf*Aef,
+             Ag*Aabf, Aabf*Aadg, Aabf*Abeg, Aabf*Acfg, Adf*Aef, Ag*Adf,
+             Aadg*Adf, Abeg*Adf, Acfg*Adf, Ag*Aef, Aadg*Aef, Abeg*Aef,
+             Acfg*Aef, Aadg*Abeg, Aadg*Acfg, Abeg*Acfg,
+             Aa + Aabf + Aace + Aadg + Aabcdefg,
+             Ab + Aabf + Abcd + Abeg + Aabcdefg,
+             Ac + Aace + Abcd + Acfg + Aabcdefg,
+             Ad + Aadg + Abcd + Ade + Adf + Aabcdefg,
+             Ae + Aace + Abeg + Ade + Aef + Aabcdefg,
+             Af + Aabf + Acfg + Adf + Aef + Aabcdefg,
+             Ag + Aadg + Abeg + Acfg + Aabcdefg]
+        """
+        flats = list(self._flats_generator)
+        reln = lambda x,y: x <= y
+        lattice_flats = Poset((flats, reln))
+        I = []
+        subsets = lattice_flats.antichains().elements_of_depth_iterator(2)
+        for subset in subsets:
+            term = poly_ring.one()
+            for el in subset:
+                term *= self._flats_generator[el]
+            I.append(term) #Stanley-Reisner Ideal
+        atoms = self._matroid.lattice_of_flats().atoms()
+        atoms_gen = {a: poly_ring.zero() for a in atoms}
+        for F in flats:
+            for a in atoms:
+                if a.issubset(F):
+                    atoms_gen[a] += self._flats_generator[F]
+        J = list(atoms_gen.values())  # Linear generators
+        return I + J
+
+    def _repr_(self):
+        r"""
+        Return a string representation of ``self``.
+
+        EXAMPLES::
+
+            sage: ch = matroids.catalog.Fano().chow_ring(QQ, False)
+            sage: ch.defining_ideal()
+            Chow ring ideal of Fano: Binary matroid of rank 3 on 7 elements,
+            type (3, 0) - non augmented
+        """
+        return "Chow ring ideal of {} - non augmented".format(self._matroid)
+
+    def _latex_(self):
+        r"""
+        Return a LaTeX representation of ``self``.
+
+        EXAMPLES::
+
+            sage: M1 = Matroid(groundset='abcd', bases=['ab','ad', 'bc'])
+            sage: ch = M1.chow_ring(QQ, False)
+            sage: ch.defining_ideal()._latex_()
+            I_{M} + J_{M} of matroid \text{\texttt{Matroid{ }of{ }rank{ }2{ }on{ }4{ }elements{ }with{ }3{ }bases}}
+        """
+        from sage.misc.latex import latex
+        return 'I_{M} + J_{M} of matroid ' + latex(self._matroid)
```

Comment:
```suggestion
        return '(I+J)_{{{}}}'.format(latex(self._matroid))
```
I think this would be better... otherwise
```suggestion
        return '(I_{{{M}}} + J_{{M{}}}'.format(M=latex(self._matroid))
```
Although this notation is out of step with the augmented cases. (Such a change would also be made in the documentation; you are free to find better notation than in the references.)